share/tools/web_config/sample_prompts/sorin.fish

```diff
@@ -1,61 +1,219 @@
-
-        # Merging state
-        test -f "$git_dir/MERGE_HEAD"
-        and printf ':'(set_color red)'merge'
-        printf ' '
-
-        # Symbols
-        if set -l count (command git rev-list --count --left-right $upstream...HEAD 2>/dev/null)
-            echo $count | read -l ahead behind
-            if test "$ahead" -gt 0
-                printf (set_color magenta)⬆' '
-            end
-            if test "$behind" -gt 0
-                printf (set_color magenta)⬇' '
-            end
+    set -l cmd_status $status
+    if test $cmd_status -ne 0
+        echo -n (set_color red)"✘ $cmd_status "
+    end
+
+    # Get the git directory for later use.
+    # Return if not inside a Git repository work tree.
+    set -l git_dir (command git rev-parse --git-dir ^&-)
+    if test $status -ne 0
+        set_color normal
+        return
+    end
+
+    # Return if zprezto's git-info is disabled.
+    set -l git_info_enabled (command git config --bool prompt.showinfo)
+    if test "$git_info_enabled" = 'false'
+        set_color normal
+        return
+    end
+
+    set -l commit ''
+    if set -l action (_git_action "$git_dir")
+        set commit (command git rev-parse --short=7 HEAD 2> /dev/null)
+    end
+
+    # Get either the branch name or a branch descriptor.
+    set -l branch_detached 0
+    if not set -l branch (command git symbolic-ref --short HEAD ^&-)
+        set branch_detached 1
+        set branch (command git describe --contains --all HEAD ^&-)
+    end
+
+    # Get the commit difference counts between local and remote.
+    set -l IFS \t
+    command git rev-list --count --left-right 'HEAD...@{upstream}' ^&- \
+        | read -l status_ahead status_behind
+    if test $status -ne 0
+        set status_ahead 0
+        set status_behind 0
+    end
+
+    # Get the stash status.
+    # (git stash list) is very slow. => Avoid using it.
+    set -l status_stashed 0
+    if test -f "$git_dir/refs/stash"
+        set status_stashed 1
+    else if test -r "$git_dir/commondir"
+        set -l IFS \n
+        read -l commondir <"$git_dir/commondir"
+        if test -f "$commondir/refs/stash"
+            set status_stashed 1
         end
+    end
 
-        for i in (git status --porcelain | string sub -l 2 | uniq)
-            switch $i
-                case "."
-                    printf (set_color green)✚' '
-                case " D"
-                    printf (set_color red)✖' '
-                case "*M*"
-                    printf (set_color blue)✱' '
-                case "*R*"
-                    printf (set_color brmagenta)➜' '
-                case "*U*"
-                    printf (set_color bryellow)═' '
-                case "??"
-                    printf (set_color brwhite)◼' '
+    # Count added, deleted, modified, renamed, unmerged, and untracked.
+    # T (type change) is undocumented, see http://git.io/FnpMGw.
+    # For a table of scenarii, see http://i.imgur.com/2YLu1.png.
```

Comment:
That's gonna bitrot.

Reply:
Do you mean the situation if imgur is gone in the future? The link itself stems from prezto's source code.
What about if I create a ASCII-style table inside the theme's source code? Would that work for you?
The resulting comment would be extremely wide though of course.

Reply:
Done. I was able to fit the whole thing within 80 characters. :+1: